app/controllers/api/v2/settings_controller.rb

```diff
@@ -27,6 +27,13 @@ def update
         process_response (@setting.parse_string_value(params[:setting][:value]) && @setting.save)
       end
 
+      api :DELETE, "/settings/:id/", N_("Reset a setting to default")
+      param :id, String, :required => true
+
+      def destroy
+        process_response (@setting.update_attribute(:value, nil))
```

Comment:
please move it to  a method on the setting model, the implementation on how to reset should be hidden in the model.